src/datahike/index/persistent_set.cljc

```diff
@@ -56,12 +56,11 @@
   (set/disj set datom (index-type->cmp-quick index-type)))
 
 (defn -insert [set datom index-type]
-  (-> (or (when-let [old (first (-slice set
-                                        (dd/datom (.-e datom) (.-a datom) (.-v datom) tx0)
-                                        (dd/datom (.-e datom) (.-a datom) (.-v datom) txmax)))]
-            (-remove set old index-type))
-          set)
-      (set/conj datom (index-type->cmp-quick index-type))))
+  (if (-slice set
+              (dd/datom (.-e datom) (.-a datom) (.-v datom) tx0)
+              (dd/datom (.-e datom) (.-a datom) (.-v datom) txmax))
+    set
+    (set/conj set datom (index-type->cmp-quick index-type))))
```

Comment:
Looks correct to me. Can you also remove the `-insert` and `-remove` duplicates (l. 34/37) though? I think they got duplicated a while ago in the course of a merge or rebase.